Makefile

```diff
@@ -187,41 +185,31 @@ build-all: $(addprefix sub-build-,$(VALIDARCHES))
 sub-build-%:
 	$(MAKE) build ARCH=$*
 
-# Update the vendored dependencies with the latest upstream versions matching
-# our glide.yaml.  If there area any changes, this updates glide.lock
-# as a side effect.  Unless you're adding/updating a dependency, you probably
-# want to use the vendor target to install the versions from glide.lock.
-.PHONY: update-vendor
-update-vendor:
-	mkdir -p $$HOME/.glide
-	$(DOCKER_RUN) $(CALICO_BUILD) glide up --strip-vendor
-	touch vendor/.up-to-date
-
 # vendor is a shortcut for force rebuilding the go vendor directory.
 .PHONY: vendor
 vendor: vendor/.up-to-date
-vendor/.up-to-date: glide.lock
-	mkdir -p $$HOME/.glide
-	$(DOCKER_RUN) $(CALICO_BUILD) glide install --strip-vendor
+vendor/.up-to-date: go.mod go.sum
+	$(DOCKER_RUN) $(CALICO_BUILD) go mod vendor
 	touch vendor/.up-to-date
 
 # Default the libcalico repo and version but allow them to be overridden
 LIBCALICO_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 LIBCALICO_REPO?=github.com/projectcalico/libcalico-go
 LIBCALICO_VERSION?=$(shell git ls-remote git@github.com:projectcalico/libcalico-go $(LIBCALICO_BRANCH) 2>/dev/null | cut -f 1)
+LIBCALICO_OLDVER?=$(shell grep libcalico-go go.mod | cut -d' ' -f2)
 
-## Update libcalico pin in glide.yaml
+## Update libcalico pin in go.mod
 update-libcalico:
 	$(DOCKER_RUN) $(CALICO_BUILD) sh -c '\
-        echo "Updating libcalico to $(LIBCALICO_VERSION) from $(LIBCALICO_REPO)"; \
-        export OLD_VER=$$(grep --after 50 libcalico-go glide.yaml |grep --max-count=1 --only-matching --perl-regexp "version:\s*\K[^\s]+") ;\
-        echo "Old version: $$OLD_VER";\
-	if [ "$(LIBCALICO_VERSION)" != $$OLD_VER ]; then \
-            sed -i "s/$$OLD_VER/$(LIBCALICO_VERSION)/" glide.yaml && \
+        echo "Updating libcalico-go to $(LIBCALICO_VERSION) from $(LIBCALICO_REPO)"; \
+        echo "Old version: $(LIBCALICO_OLDVER)";\
+	if [ "$(LIBCALICO_VERSION)" != "$(LIBCALICO_OLDVER)" ]; then \
+	    sed -i "/libcalico-go/d" go.mod go.sum; \
+	    go get $(LIBCALICO_REPO)@$(LIBCALICO_VERSION); \
 	    if [ "$(LIBCALICO_REPO)" != "github.com/projectcalico/libcalico-go" ]; then \
-              glide mirror set https://github.com/projectcalico/libcalico-go $(LIBCALICO_REPO) --vcs git; glide mirror list; \
+	      echo "replace github.com/projectcalico/libcalico-go => $(LIBCALICO_REPO) v0.0.0-$(LIBCALICO_VERSION)" >> go.mod; \
```

Comment:
I don't think you can hardcode `v0.0.0-` into the version.  I just ran the update to `68e175bba1ebf524e54a4d0034119e4cf146bffb` and got the resulting version:

```
github.com/projectcalico/libcalico-go v3.9.0-0.dev.0.20190801150727-68e175bba1eb+incompatible
```

Reply:
I'm replacing that "echo" command with 
```go mod edit -replace github.com/projectcalico/libcalico-go=$(LIBCALICO_REPO)@$(LIBCALICO_VERSION)```
I think that should do it w/o all the v0.0.0 problems.